File: app/src/main/java/app/we/go/emojidraw/features/practice/EmojiDetectedViewHolder.java
package app.we.go.emojidraw.features.practice;

import android.animation.ObjectAnimator;
import android.animation.PropertyValuesHolder;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

class EmojiDetectedViewHolder extends RecyclerView.ViewHolder {


    private final TextView emojiView;
    private final ObjectAnimator pulseAnimation;

    EmojiDetectedViewHolder(View itemView) {
        super(itemView);
        emojiView = (TextView) itemView;

        pulseAnimation = ObjectAnimator.ofPropertyValuesHolder(
                emojiView,
                PropertyValuesHolder.ofFloat("scaleX", 1.2f),
                PropertyValuesHolder.ofFloat("scaleY", 1.2f));
        pulseAnimation.setDuration(310);

        pulseAnimation.setRepeatCount(ObjectAnimator.INFINITE);
        pulseAnimation.setRepeatMode(ObjectAnimator.REVERSE);
    }


    void bind(String emoji, boolean isTheOne) {
        emojiView.setText(emoji);

        // show a pulse animation on the element that represents the correct emoji
        if (isTheOne) {
            pulseAnimation.start();
        } else {
            pulseAnimation.cancel();
        }
    }
}

File: app/src/main/java/app/we/go/emojidraw/features/practice/EmojiDetectedViewHolder.kt
package app.we.go.emojidraw.features.practice

import android.animation.ObjectAnimator
import android.animation.PropertyValuesHolder
import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class EmojiDetectedViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {


    private val emojiView: TextView = itemView as TextView
    private val pulseAnimation: ObjectAnimator

    init {
        pulseAnimation = ObjectAnimator.ofPropertyValuesHolder(
                emojiView,
                PropertyValuesHolder.ofFloat("scaleX", 1.2f),
                PropertyValuesHolder.ofFloat("scaleY", 1.2f))
        pulseAnimation.duration = 310

        pulseAnimation.repeatCount = ObjectAnimator.INFINITE
        pulseAnimation.repeatMode = ObjectAnimator.REVERSE
    }


    fun bind(emoji: String, isTheOne: Boolean) {
        emojiView.text = emoji

        // show a pulse animation on the element that represents the correct emoji
        if (isTheOne) {
            pulseAnimation.start()
        } else {
            pulseAnimation.cancel()
        }
    }
}
